skbio/diversity/beta/tests/test_unifrac_base.py

```diff
@@ -0,0 +1,612 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+from io import StringIO
+
+import numpy as np
+
+from skbio import TreeNode
+from skbio.tree import DuplicateNodeError, MissingNodeError
+
+
+class StatsTests(object):
+    _method = {}
+
+    def setUp(self):
+        self.unweighted_unifrac = self._method['unweighted_unifrac']
```

Comment:
As a side comment, in [subsample](https://github.com/biocore/scikit-bio/blob/master/skbio/stats/tests/test_subsample.py#L126-L133) there's a pretty neat way to test several implementations against the same test cases.
It is the exact same pattern that CPython itself [uses](https://github.com/python/cpython/search?utf8=%E2%9C%93&q=import_fresh_module) to test accelerated modules.